routes/api.js

```diff
@@ -228,16 +228,25 @@ router.post('/request_sms', apiMiddleware(async (req) => {
     throw new ApiError({ field: 'prefix', type: 'error_api_prefix_invalid' });
   }
 
-  let phoneNumber;
+  let phoneNumber = phoneUtil.parse(req.body.phoneNumber, countryCode);
+
+  const isValid = phoneUtil.isValidNumber(phoneNumber);
+
+  if (!isValid) {
+    throw new ApiError({ field: 'phoneNumber', type: 'error_phone_invalid' });
+  }
+
+  phoneNumber = phoneUtil.format(
+    phoneNumber,
+    PNF.E164,
+  );
+
   try {
-    phoneNumber = phoneUtil.format(
-      phoneUtil.parse(req.body.phoneNumber, countryCode),
-      PNF.INTERNATIONAL,
-    );
-    phoneNumber = phoneNumber.replace(/[^+0-9]+/g, '');
```

Comment:
Is the PNF.E164 formatter guaranteed to output numbers w/o whitespace? Conveyor will not accept any whitespace in numbers when transferring the data

Reply:
yes.

`INTERNATIONAL` includes whitespace ... and this regexp is basically turning `INTERNATIONAL` into `E164`...

Reply:
👌